x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/DeprecationInfoAction.java

```diff
@@ -211,6 +211,14 @@ public int hashCode() {
                 }
             }
 
+            // WORKAROUND: move transform deprecation issues into cluster_settings
```

Comment:
The only question I have is whether @jakelandis or @masseyke would like a reference to an issue here, as I believe the plan is that there might be more categories of deprecations allowed in the future, and if/when that happens it would obviously be good to refactor this.  If there is no GitHub issue for the future work then I guess at least `git blame` will lead back to this PR during the future refactoring.

Reply:
We had discussed changing the categories but decided to against it in for 7.x. It is a possibility for 8.x+, but we are still early  in the discussions and will be focusing more on re-evaluating the UX, for which better categories might be path, but not certain yet.

Reply:
:+1: that justifies this workaround even more. It's easy to remove/re-factor after you've decided the new categories. 

Do you have a tracking issue number I can add to the code comment?

Reply:
No, we don't have a tracking issue yet. Without a better sense of direction it is likely too early.